Automattic/newspack-listings
# refactor: use AutocompleteWithSuggestions from newspack-components

### All Submissions:






### Changes proposed in this Pull Request:



A refactor use the new `AutocompleteWithSuggestions` component from `newspack-components` in place of the custom/internal `QueryControls` component (which was more or less the same thing).

Also makes the block appender button in the Curated List block always visible, instead of visible only when the block is selected, as recent changes in Gutenberg have made it frustratingly difficult to select the wrapping Curated List block (either through the block selector menu or by clicking on it directly). This is a temporary solution to feedback from many publishers that it seems nearly impossible to add new listing blocks to a Specific Listings-mode Curated List. Open to alternative UI improvements to address this (cc @thomasguillot).

### How to test the changes in this Pull Request:

There shouldn't be any functional changes other than the block appender being always visible. To test:

1. Create a Curated List block in a post or page, select "Specific Listings" mode
2. Confirm that the [ + ] appender button is always visible and clickable to add new items to the list
3. Click the appender button and add a listing of any type
4. Confirm that the placeholder autocomplete/search/select functionality works as expected

### Other information:
